### PR TITLE
fix: token icon not show on BSC

### LINKED ISSUE #3305: [Bug] Fix the token icon url on BSC
### How to reproduce?

Example:

https://exchange.pancakeswap.finance/images/coins/0x8f0528ce5ef7b51152a59745befdd91d97091d2f.png

### Mask Version:

- [ ] (Browser) Installed from the extension store (version: )
- [ ] (App) Installed from the app store (version: )
- [ ] master branch
- [ ] Build of version/commit hash:

(Please write down build command/flags if there is anything special)

### System

 
- [ ] Windows 10
- [ ] Mac OS X Catalina
- [ ] Linux: 

### Platform/Browser

 
- [ ] Latest Chrome (stable channel)
- [ ] Latest Firefox (stable channel)
- [ ] Android 11
- [ ] iOS 14

### PR DESCRIPTION
**Pull request by NikoManiac**
closes #3305

### CODE REVIEW

File: packages/maskbook/src/extension/options-page/DashboardComponents/TokenIcon.tsx [hunk 1]
```diff
@@ -5,27 +5,61 @@ import {
...
-            'https://dimensiondev.github.io/Maskbook-VI/assets/Logo/MB--Logo--Geo--ForceCircle--Blue.svg', // MASK
-        '0x32a7C02e79c4ea1008dD6564b35F131428673c41': 'https://s2.coinmarketcap.com/static/img/coins/64x64/6747.png', // CRUST
-        '0x04abEdA201850aC0124161F037Efd70c74ddC74C': 'https://s2.coinmarketcap.com/static/img/coins/64x64/5841.png', // NEST
-        '0x14de81C71B3F73874659082b971433514E201B27': 'https://etherscan.io/token/images/ykyctoken_32.png', // Yes KYC
-        '0x3B73c1B2ea59835cbfcADade5462b6aB630D9890':
-            'https://raw.githubusercontent.com/chainswap/chainswap-assets/main/logo_white_256.png', // TOKEN
+function resolveTokenIconURLs(address: string, baseURIs: string[], chainId: ChainId, logoURI?: string) {
+    const checkSummedAddress = formatEthereumAddress(address)
+
+    if (constantOfChain(CONSTANTS, chainId).NATIVE_TOKEN_ADDRESS === checkSummedAddress) {
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/options-page/DashboardComponents/TokenIcon.tsx:55]: Use `isSameAddress` to compare addresses.

**Review by guanbinrui — APPROVED** [review]

**Review by UncleBill — APPROVED** [review]

**Review by UncleBill — DISMISSED** [review]

File: packages/maskbook/src/extension/options-page/DashboardComponents/TokenIcon.tsx [hunk 2]
```diff
@@ -5,27 +5,62 @@ import {
...
+    useConstant,
+    constantOfChain,
+    isSameAddress,
 } from '@dimensiondev/web3-shared'
 import { useStylesExtends } from '../../../components/custom-ui-helper'
-import { useImageFailover } from '../../../utils/hooks/useImageFailover'
+import { useImageFailover } from '../../../utils'
+
+const SPECIAL_ICON_ASSET_MAP: { [key: string]: string } = {
+    '0x04abEdA201850aC0124161F037Efd70c74ddC74C': 'https://s2.coinmarketcap.com/static/img/coins/64x64/5841.png', // NEST
```
Comment (septs) [inline packages/maskbook/src/extension/options-page/DashboardComponents/TokenIcon.tsx:16]: why not is json file?

Comment (septs) [reply packages/maskbook/src/extension/options-page/DashboardComponents/TokenIcon.tsx:16]: current structure not is semantic
```json
[
    {
         "name": "",
         "address": ".....",
         "logo_url": "...."
    }
]
```

File: packages/maskbook/src/extension/options-page/DashboardComponents/TokenIcon.tsx [hunk 3]
```diff
@@ -40,41 +41,25 @@ const useStyles = makeStyles((theme: Theme) => ({
 
 export interface TokenIconProps extends withClasses<never> {
     name?: string
-    logoURL?: string
+    logoURI?: string
```
Comment (septs) [inline packages/maskbook/src/extension/options-page/DashboardComponents/TokenIcon.tsx:44]: why modify this field name?

**Review by guanbinrui — DISMISSED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**